File: src/main/java/anthill/Anthill/db/repository/MemberRepository.java
package anthill.Anthill.db.repository;

import anthill.Anthill.db.domain.member.Member;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long> {
    Optional<Member> findByUserId(String userId);

    boolean existsByUserId(String userId);

    boolean existsByNickName(String nickName);

    boolean existsByPhoneNumber(String phoneNumber);
}

File: src/main/kotlin/anthill/Anthill/db/repository/MemberRepository.kt
package anthill.Anthill.db.repository

import anthill.Anthill.db.domain.member.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {
    fun findByUserId(userId: String): Optional<Member>
    fun existsByUserId(userId: String): Boolean
    fun existsByNickName(nickName: String): Boolean
    fun existsByPhoneNumber(phoneNumber: String): Boolean
}